### PR TITLE
Fix/mathjax provider just for mathjax lessons LESQ-774

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: {owa_music_year}

- Limits lesson overview page mathjax usage - no legacy subject and only from allowed list if they contain mathjax in lesson data.

Allowed subject slugs:

  "maths",
  "physics",
  "chemistry",
  "biology",
  "combined-science",
  "science",

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2365--oak-web-application.netlify.thenational.academy
Check allowed lessons with MathJax
Type window.MathJax into the console and press Enter. If MathJax is loaded, you should see an object returned that represents the MathJax library. If it returns undefined, then MathJax is not loaded on the page. You may need to clear the cache between tests

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherViews/LessonOverview/LessonOverview.view.tsx [hunk 1]
```diff
@@ -102,6 +107,18 @@ export function LessonOverview({ lesson }: LessonOverviewProps) {
...
   } = commonPathway;
 
+  const isLegacyLicense = !lessonCohort || lessonCohort === LEGACY_COHORT;
+  const isNew = lessonCohort === NEW_COHORT;
+
+  const isMathJaxLesson =
+    subjectSlug &&
+    ALLOWED_MATHJAX_SUBJECT_SLUGS.includes(subjectSlug) &&
+    !isLegacyLicense
+      ? hasLessonMathJax(lesson)
```
Comment (njblore) [inline src/components/TeacherViews/LessonOverview/LessonOverview.view.tsx:117]: can hasLessonMathjax do all these checks internally? would be cleaner than having the component have to know about all these conditions

File: src/components/TeacherComponents/LessonOverviewDetails/LessonOverviewDetails.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ type LessonOverviewDetailsProps = {
   contentGuidance: ContentGuidance[] | null | undefined;
   supervisionLevel: string | null | undefined;
   isLegacyLicense?: boolean;
+  isMathJaxLesson?: boolean;
```
Comment (njblore) [inline src/components/TeacherComponents/LessonOverviewDetails/LessonOverviewDetails.tsx:33]: does this need to be optional?

File: src/components/TeacherViews/LessonOverview/hasLessonMathJax.test.ts [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    expect(hasLessonMathJax(lessonWithMathJaxInExitQuiz)).toBe(true);
+  });
+
+  it("detects MathJax in starterQuiz questions", () => {
+    const lessonWithMathJaxInStarterQuiz: LessonOverviewProps["lesson"] = {
+      ...basicLesson,
+      starterQuiz: [
+        {
+          questionId: 1, // replace with actual questionId
+          questionUid: "example-uid", // replace with actual questionUid
```
Comment (njblore) [inline src/components/TeacherViews/LessonOverview/hasLessonMathJax.test.ts:258]: are these comments out dated?

File: src/components/TeacherViews/LessonOverview/LessonOverview.view.tsx (hunk 1 above)
Comment (JBR90) [reply src/components/TeacherViews/LessonOverview/LessonOverview.view.tsx:117]: Yes could do, would need to pass lesson, subjectSlug and isLegacyLicense and do the ALLOWED_MATHJAX_SUBJECT_SLUGS check in the function. is that what you mean ?

Comment (njblore) [reply src/components/TeacherViews/LessonOverview/LessonOverview.view.tsx:117]: yup exactly, can just extract this exact piece of code into another function

File: src/components/TeacherComponents/LessonOverviewDetails/LessonOverviewDetails.tsx (hunk 2 above)
Comment (JBR90) [reply src/components/TeacherComponents/LessonOverviewDetails/LessonOverviewDetails.tsx:33]: no

**Review by njblore — APPROVED** [review]
🙌

**Review by jack-skerrett-oak — APPROVED** [review]
Mathjax seems to be rendering as before, and I can't see the library loading on pages where mathjax isn't present (even inside list the of subjects to check)

**Pull request merged**